### DISCUSSION TITLE
LOTW and FT4

### DISCUSSION THREAD

**Discussion by kj4z** [magicbug/Cloudlog#954] [Q&A]
I noticed that none of my FT4 QSOs are appearing on LoTW.  According to the ADIF spec, it would seem that the correct reporting format is mode=MFSK, submode=FT4.  However, even when I manually modified the view to format the ADIF accordingly, LOTW still silently dropped those QSOs.

What did work is to simply override MFSK as the mode and set it to FT4 when uploading to LOTW.

Is anyone else seeing this issue?  I am happy to submit a pull request to address it if this was not a previously known issue.  Closely related to https://github.com/magicbug/Cloudlog/discussions/948

**Comment by dg9vh** [comment]
I also can reproduce this on my log - did not recognized it till you pushed me to it :-)

A PR would solve this problem as you said... feel free to push.

**Comment by magicbug** [comment]
This one's a bit harder to do with remap although simply remapping mfsk to ft4 would break it for other modes.

Might be that I need to check for submode and if its present use that, I'll try do a little digging tomorrow

**Comment by magicbug** [comment]
Probably worth putting this into an issue as well as its not a bug as things will match but nice to get sub mode included

**Reply by kj4z** [reply]
I've created a PR which I believe should be easy to maintain going forward, and I have validated that LOTW accepts both FT4 and FT8 QSOs generated by it.  Additionally, LOTW confirmations for FT4 QSOs are correctly reflected in the local database as expected.  It's been a long time since I used CodeIgniter, so feedback welcome.

**Reply by magicbug** [reply]
I'm just heading to bed but I'll get it checked over tomorrow afternoon

**Comment by magicbug** [comment]
@kj4z Looks great thanks for adding the changes, merged into the main branch of Cloudlog.

I'm going to take it FT-8 is uploading OK?

**Reply by kj4z** [reply]
Yeah, FT8 has always worked because it's considered its own mode instead of a submode, for whatever reason.

Cloudlog is looking great these days.  Thanks for maintaining.

**Reply by magicbug** [reply]
Thanks, I try, still tons todo and improve as with any project
